api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -106,6 +133,16 @@ private String convertClusterName(String workspaceId) {
 
   public ResponseEntity<Cluster> getCluster(String workspaceNamespace,
       String workspaceId) {
+
+    // This also enforces registered auth domain.
+    WorkspaceAccessLevel accessLevel;
+    try {
+      accessLevel = workspaceService.enforceWorkspaceAccessLevel(workspaceNamespace,
+          workspaceId, WorkspaceAccessLevel.WRITER);
+    } catch (Exception e) {
```

Comment:
In general: we shouldn't catch Exception.

And there's no reason to catch e and then throw it. You can remove these try / catch blocks throughout?